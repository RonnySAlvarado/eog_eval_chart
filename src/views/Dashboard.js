import React, { useState } from 'react';
import Loader from 'react-loader-spinner';

import GraphSelectionButton from '../components/GraphSelectionButton';
import Linegraph from '../components/Linegraph';
import { DashboardHeaderContainer } from '../styles/DashboardHeaderContainer';
import { LoaderContainer } from '../styles/LoaderContainer';
import { DashboardContainer } from '../styles/DashboardContainer';
import useGetMeasurements from '../hooks/useGetMeasurements';

const Dashboard = () => {
  const [loading, error, data] = useGetMeasurements();
  const metricList = ['waterTemp', 'casingPressure', 'injValveOpen', 'flareTemp', 'oilTemp', 'tubingPressure'];
  const [view, setView] = useState([
    'waterTemp',
    'casingPressure',
    'injValveOpen',
    'flareTemp',
    'oilTemp',
    'tubingPressure',
  ]);

  if (error) {
    return <p>Error: {error}</p>;
  }

  if (loading || data.getMultipleMeasurements === undefined) {
    return (
      <LoaderContainer>
        <Loader type="BallTriangle" color="#00BFFF" height={100} width={100} />
      </LoaderContainer>
    );
  } else {
    return (
      <DashboardContainer>
        <DashboardHeaderContainer>
          {metricList.map((type, id) => {
            let value =
              data.getMultipleMeasurements[id].measurements[data.getMultipleMeasurements[id].measurements.length - 1]
                .value;
            return <GraphSelectionButton key={id} type={type} value={value} setView={setView} view={view} />;
          })}
        </DashboardHeaderContainer>
        <Linegraph data={data} view={view} />
      </DashboardContainer>
    );
  }
};

export default Dashboard;
