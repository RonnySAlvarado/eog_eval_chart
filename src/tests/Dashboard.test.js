import React from 'react';
import { render } from '@testing-library/react';
import { ApolloProvider } from '@apollo/react-hooks';
import client from '../utils/createClient';

import Dashboard from '../views/Dashboard';

describe('<Dashboard />', () => {
  it('Matches snapshot', () => {
    const dashboard = render(
      <ApolloProvider client={client}>
        <Dashboard />
      </ApolloProvider>,
    );
    expect(dashboard).toMatchSnapshot();
  });
});
