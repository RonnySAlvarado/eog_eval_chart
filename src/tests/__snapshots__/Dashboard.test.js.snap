// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> Matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-AykKD hXrZrR"
      >
        <div
          aria-busy="true"
          class=""
        >
          <svg
            aria-label="audio-loading"
            height="100"
            stroke="#00BFFF"
            viewBox="0 0 57 57"
            width="100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fill-rule="evenodd"
            >
              <g
                stroke-width="2"
                transform="translate(1 1)"
              >
                <circle
                  cx="5"
                  cy="50"
                  r="5"
                >
                  <animate
                    attributeName="cy"
                    begin="0s"
                    calcMode="linear"
                    dur="2.2s"
                    repeatCount="indefinite"
                    values="50;5;50;50"
                  />
                  <animate
                    attributeName="cx"
                    begin="0s"
                    calcMode="linear"
                    dur="2.2s"
                    repeatCount="indefinite"
                    values="5;27;49;5"
                  />
                </circle>
                <circle
                  cx="27"
                  cy="5"
                  r="5"
                >
                  <animate
                    attributeName="cy"
                    begin="0s"
                    calcMode="linear"
                    dur="2.2s"
                    from="5"
                    repeatCount="indefinite"
                    to="5"
                    values="5;50;50;5"
                  />
                  <animate
                    attributeName="cx"
                    begin="0s"
                    calcMode="linear"
                    dur="2.2s"
                    from="27"
                    repeatCount="indefinite"
                    to="27"
                    values="27;49;5;27"
                  />
                </circle>
                <circle
                  cx="49"
                  cy="50"
                  r="5"
                >
                  <animate
                    attributeName="cy"
                    begin="0s"
                    calcMode="linear"
                    dur="2.2s"
                    repeatCount="indefinite"
                    values="50;50;5;50"
                  />
                  <animate
                    attributeName="cx"
                    begin="0s"
                    calcMode="linear"
                    dur="2.2s"
                    from="49"
                    repeatCount="indefinite"
                    to="49"
                    values="49;5;27;49"
                  />
                </circle>
              </g>
            </g>
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AykKD hXrZrR"
    >
      <div
        aria-busy="true"
        class=""
      >
        <svg
          aria-label="audio-loading"
          height="100"
          stroke="#00BFFF"
          viewBox="0 0 57 57"
          width="100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fill-rule="evenodd"
          >
            <g
              stroke-width="2"
              transform="translate(1 1)"
            >
              <circle
                cx="5"
                cy="50"
                r="5"
              >
                <animate
                  attributeName="cy"
                  begin="0s"
                  calcMode="linear"
                  dur="2.2s"
                  repeatCount="indefinite"
                  values="50;5;50;50"
                />
                <animate
                  attributeName="cx"
                  begin="0s"
                  calcMode="linear"
                  dur="2.2s"
                  repeatCount="indefinite"
                  values="5;27;49;5"
                />
              </circle>
              <circle
                cx="27"
                cy="5"
                r="5"
              >
                <animate
                  attributeName="cy"
                  begin="0s"
                  calcMode="linear"
                  dur="2.2s"
                  from="5"
                  repeatCount="indefinite"
                  to="5"
                  values="5;50;50;5"
                />
                <animate
                  attributeName="cx"
                  begin="0s"
                  calcMode="linear"
                  dur="2.2s"
                  from="27"
                  repeatCount="indefinite"
                  to="27"
                  values="27;49;5;27"
                />
              </circle>
              <circle
                cx="49"
                cy="50"
                r="5"
              >
                <animate
                  attributeName="cy"
                  begin="0s"
                  calcMode="linear"
                  dur="2.2s"
                  repeatCount="indefinite"
                  values="50;50;5;50"
                />
                <animate
                  attributeName="cx"
                  begin="0s"
                  calcMode="linear"
                  dur="2.2s"
                  from="49"
                  repeatCount="indefinite"
                  to="49"
                  values="49;5;27;49"
                />
              </circle>
            </g>
          </g>
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
